spec/requests/api/v2/sample_metadata_spec.rb

```diff
@@ -0,0 +1,199 @@
+
+          expect(response).to have_http_status(:success)
+          expect(json.dig('data', 'attributes', 'cohort')).to eq 'updated cohort'
+          expect(json.dig('data', 'attributes', 'collected_by')).to eq 'updated collected_by'
+          expect(json.dig('data', 'attributes', 'concentration')).to eq 'updated concentration'
+          expect(json.dig('data', 'attributes', 'donor_id')).to eq 'updated donor_id'
+          expect(json.dig('data', 'attributes', 'gender')).to eq 'Female'
+          expect(json.dig('data', 'attributes', 'sample_common_name')).to eq 'updated sample_common_name'
+          expect(json.dig('data', 'attributes', 'sample_description')).to eq 'updated sample_description'
+          expect(json.dig('data', 'attributes', 'supplier_name')).to eq 'updated supplier_name'
+          expect(json.dig('data', 'attributes', 'volume')).to eq 'updated volume'
+        end
+
+        it 'updates the model correctly' do
+          # Apply the patch which replaced all the metadata
+          api_patch "#{base_endpoint}/#{resource_model.id}", payload
+
+          # Check that the model was modified
+          resource_model.reload
+          expect(resource_model.cohort).to eq 'updated cohort'
+          expect(resource_model.collected_by).to eq 'updated collected_by'
+          expect(resource_model.concentration).to eq 'updated concentration'
+          expect(resource_model.donor_id).to eq 'updated donor_id'
+          expect(resource_model.gender).to eq 'Female'
+          expect(resource_model.sample_common_name).to eq 'updated sample_common_name'
+          expect(resource_model.sample_description).to eq 'updated sample_description'
+          expect(resource_model.supplier_name).to eq 'updated supplier_name'
+          expect(resource_model.volume).to eq 'updated volume'
+        end
+      end
+
+      context 'with a missing type in the payload' do
+        let(:payload) { { 'data' => { 'id' => resource_model.id, 'attributes' => { cohort: 'updated cohort' } } } }
+
+        it 'does not update the collection' do
+          original_cohort = resource_model.cohort
+
+          api_patch "#{base_endpoint}/#{resource_model.id}", payload
+          expect(response).to have_http_status(:bad_request)
+          expect(json['errors'][0]['title']).to eq('Missing Parameter')
+
+          # Check that the model was not modified
+          resource_model.reload
+          expect(resource_model.cohort).to eq original_cohort
+        end
+      end
+    end
+
+    describe '#post' do
+      context 'with a valid payload' do
+        let(:payload) do
+          {
+            'data' => {
+              'type' => 'sample_metadata',
+              'attributes' => {
+                cohort: 'posted cohort',
+                collected_by: 'posted collected_by',
+                concentration: 'posted concentration',
+                donor_id: 'posted donor_id',
+                gender: 'mixed',
```

Comment:
can you make that 'posted gender' so it matches the others? confused me

Reply:
or is it an ENUM or something fixed?

Reply:
Yeah it's an enum otherwise I would have done